File: Web/src/api/import/importTask/wmsImportTask.ts
import {useBaseApi} from '/@/api/base';

// 入库任务表接口服务
export const useWmsImportTaskApi = () => {
	const baseApi = useBaseApi("wmsImportTask");
	return {
		// 分页查询入库任务表
		page: baseApi.page,
		// 查看入库任务表详细
		detail: baseApi.detail,
		// 新增入库任务表
		add: baseApi.add,
		// 更新入库任务表
		update: baseApi.update,
		// 删除入库任务表
		delete: baseApi.delete,
		// 批量删除入库任务表
		batchDelete: baseApi.batchDelete,
		// 导出入库任务表数据
		exportData: baseApi.exportData,
		// 导入入库任务表数据
		importData: baseApi.importData,
		// 下载入库任务表数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
		// 取消入库任务
        cancelTask: baseApi.cancelTask,
	}
}

// 入库任务表实体
export interface WmsImportTask {
	// 主键Id
	id: number;
	// 任务号
	taskNo: string;
	// 发送方
	sender: string;
	// 接收方
	receiver: string;
	// 是否成功
	isSuccess: number;
	// 发送时间
	sendDate: string;
	// 结束时间
	backDate: string;
	// 描述发送报文
	message: string;
	// 托盘码
	stockCode: string;
	// 描述返回信息
	msg: string;
	// 任务状态
	status: number;
	// 取消时间
	cancelDate: string;
	// 完成时间
	finishDate: string;
	// 仓库ID
	wareHouseId: number;
	// 起始位置
	startLocation: string;
	// 目标位置
	endLocation: string;
	// 二次申请后目标位置
	newEnd: string;
	// 创建者姓名
	createUserName: string;
	// 修改者姓名
	updateUserName: string;
}

File: Admin.NET/Admin.NET.Application/Entity/WmsImportTask.cs
// Admin.NET 项目的版权、商标、专利和其他相关权利均受相应法律法规的保护。使用本项目应遵守相关法律法规和许可证的要求。
//
// 本项目主要遵循 MIT 许可证和 Apache 许可证（版本 2.0）进行分发和使用。许可证位于源代码树根目录中的 LICENSE-MIT 和 LICENSE-APACHE 文件。
//
// 不得利用本项目从事危害国家安全、扰乱社会秩序、侵犯他人合法权益等法律法规禁止的活动！任何基于本项目二次开发而产生的一切法律纠纷和责任，我们不承担任何责任！

using Admin.NET.Core;
using SqlSugar;
namespace Admin.NET.Application.Entity;

/// <summary>
/// 入库任务表
/// </summary>
[Tenant("1300000000001")]
[SugarTable("WmsImportTask", "入库任务表")]
public partial class WmsImportTask : EntityBaseDel
{
    /// <summary>
    /// 任务号
    /// </summary>
    [SugarColumn(ColumnName = "TaskNo", ColumnDescription = "任务号", Length = 255)]
    public virtual string? TaskNo { get; set; }
    
    /// <summary>
    /// 发送方
    /// </summary>
    [SugarColumn(ColumnName = "Sender", ColumnDescription = "发送方", Length = 255)]
    public virtual string? Sender { get; set; }
    
    /// <summary>
    /// 接收方
    /// </summary>
    [SugarColumn(ColumnName = "Receiver", ColumnDescription = "接收方", Length = 255)]
    public virtual string? Receiver { get; set; }
    
    /// <summary>
    /// 是否成功
    /// </summary>
    [SugarColumn(ColumnName = "IsSuccess", ColumnDescription = "是否成功")]
    public virtual int? IsSuccess { get; set; }
    
    /// <summary>
    /// 发送时间
    /// </summary>
    [SugarColumn(ColumnName = "SendDate", ColumnDescription = "发送时间")]
    public virtual DateTime? SendDate { get; set; }
    
    /// <summary>
    /// 结束时间
    /// </summary>
    [SugarColumn(ColumnName = "BackDate", ColumnDescription = "结束时间")]
    public virtual DateTime? BackDate { get; set; }
    
    /// <summary>
    /// 描述发送报文
    /// </summary>
    [SugarColumn(ColumnName = "Message", ColumnDescription = "描述发送报文", Length = 255)]
    public virtual string? Message { get; set; }
    
    /// <summary>
    /// 托盘码
    /// </summary>
    [SugarColumn(ColumnName = "StockCode", ColumnDescription = "托盘码", Length = 255)]
    public virtual string? StockCode { get; set; }
    
    /// <summary>
    /// 描述返回信息
    /// </summary>
    [SugarColumn(ColumnName = "Msg", ColumnDescription = "描述返回信息", Length = 255)]
    public virtual string? Msg { get; set; }
    
    /// <summary>
    /// 任务状态
    /// </summary>
    [SugarColumn(ColumnName = "Status", ColumnDescription = "任务状态")]
    public virtual int? Status { get; set; }
    
    /// <summary>
    /// 取消时间
    /// </summary>
    [SugarColumn(ColumnName = "CancelDate", ColumnDescription = "取消时间")]
    public virtual DateTime? CancelDate { get; set; }
    
    /// <summary>
    /// 完成时间
    /// </summary>
    [SugarColumn(ColumnName = "FinishDate", ColumnDescription = "完成时间")]
    public virtual DateTime? FinishDate { get; set; }
    
    /// <summary>
    /// 仓库ID
    /// </summary>
    [SugarColumn(ColumnName = "WareHouseId", ColumnDescription = "仓库ID")]
    public virtual long? WareHouseId { get; set; }
    
    /// <summary>
    /// 起始位置
    /// </summary>
    [SugarColumn(ColumnName = "StartLocation", ColumnDescription = "起始位置", Length = 50)]
    public virtual string? StartLocation { get; set; }
    
    /// <summary>
    /// 目标位置
    /// </summary>
    [SugarColumn(ColumnName = "EndLocation", ColumnDescription = "目标位置", Length = 50)]
    public virtual string? EndLocation { get; set; }
    
    /// <summary>
    /// 二次申请后目标位置
    /// </summary>
    [SugarColumn(ColumnName = "NewEnd", ColumnDescription = "二次申请后目标位置", Length = 32)]
    public virtual string? NewEnd { get; set; }
    
}
